external-api/src/http/wallet.rs

```diff
@@ -152,6 +152,16 @@ pub struct CreateOrderRequest {
     /// The authorization parameters for the update
     #[serde(flatten)]
     pub update_auth: WalletUpdateAuthorization,
+    /// The options for creating the order
+    #[serde(default)]
+    pub options: CreateOrderOptions,
+}
+
+/// The options for creating an order
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+pub struct CreateOrderOptions {
```

Comment:
should we move the `matching_pool` option into this struct? seems like a natural place for it to be